sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubAsyncProducer.java

```diff
@@ -354,11 +354,16 @@
                         .setPartitionKey(partitionKey)
                         .setMaximumSizeInBytes(batchSize);
 
+                    final AtomicReference<Boolean> isFirst = new AtomicReference<>(true);
```

Comment:
AtomicBoolean is sufficient. Then you can do:

```java
if (!isFirst.getAndSet(true)) {
    // update sendSpanContext only once

    Context entityContext = parentContext.addData(ENTITY_PATH, link.getEntityPath()); 
    sendSpanContext.set(tracerProvider 
        .startSpan(entityContext.addData(HOST_NAME, link.getHostname()), ProcessKind.SEND));
}

// add span context on event data
return setSpanContext(eventData, parentContext);
```

Reply:
Would suggest writing a test for this, too.

Reply:
updated, with the above code and fixed the already existing test cases for this.